File: src/main/java/esprit/example/fegaac1/entities/Reclamation.java
package esprit.example.fegaac1.entities;


import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.Set;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Reclamation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id_reclamation;

    private String description;
    private String titre;
    private String localisation ;

    @Enumerated(EnumType.STRING)
    private ETAT_RECLAMATION etatReclamation = ETAT_RECLAMATION.EN_ATTENTE;

    private LocalDate date = LocalDate.now();

    @ManyToOne
    @JoinColumn(name = "user_id")
    @JsonIgnoreProperties({"reclamations"})
    private User user;

}

File: src/app/models/Reclamation.ts
import { ETAT_RECLAMATION } from './ETAT_RECLAMATION';
import { User } from './user';

export interface Reclamation {
  id_reclamation?: number;
  titre: string;
  description: string;
  localisation: string;
  etatReclamation?: ETAT_RECLAMATION;
  date?: string; 
   user?: User;
}
